File: api/me.js
export default async function handler(req, res) {
  // Set CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    res.status(200).end();
    return;
  }

  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  try {
    const { access_token } = req.body;

    const headers = {
      'Authorization': `Bearer ${access_token}`
    };

    const response = await fetch('https://api.spotify.com/v1/me', {
      headers: headers
    });

    if (!response.ok) {
      const errorData = await response.json();
      return res.status(400).json({
        error: 'Failed to fetch user profile',
        details: errorData
      });
    }

    const userData = await response.json();
    return res.status(200).json(userData);
  } catch (error) {
    console.error('Error in me:', error);
    return res.status(500).json({
      error: 'Internal server error',
      details: error.message
    });
  }
}

File: api/me.py
from flask import Flask, request, jsonify
import requests
from dotenv import load_dotenv

load_dotenv()

app = Flask(__name__)

@app.route("/api/me", methods=["POST"])
def get_user_profile():
    data = request.json
    access_token = data.get("access_token")

    headers = {
        "Authorization": f"Bearer {access_token}"
    }

    r = requests.get("https://api.spotify.com/v1/me", headers=headers)

    if r.status_code != 200:
        return jsonify({
            "error": "Failed to fetch user profile",
            "details": r.json()
        }), 400

    return jsonify(r.json())

def handler(request):
    return app(request.environ, request.start_response)